demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,390 @@
+
+#include <inference_engine.hpp>
+
+#include <format_reader_ptr.h>
+
+#include <samples/common.hpp>
+#include <samples/slog.hpp>
+#include <samples/args_helper.hpp>
+#include <samples/classification_results.h>
+#include <samples/ocv_common.hpp>
+#include <samples/common.hpp>
+
+#include <vpu/vpu_plugin_config.hpp>
+#include <cldnn/cldnn_config.hpp>
+
+#include <opencv2/core.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/highgui.hpp>
+
+#include <sys/stat.h>
+
+#include "imagenet_classification_demo.hpp"
+#include "grid_mat.hpp"
+#include "infer_request_callback.hpp"
+
+ConsoleErrorListener error_listener;
+
+bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+    // ---------------------------------Parsing and validation of input args----------------------------------
+    slog::info << "Parsing input parameters" << slog::endl;
+
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_h) {
+        showUsage();
+        showAvailableDevices();
+        return false;
+    }
+    slog::info << "Parsing input parameters" << slog::endl;
+
+    if (FLAGS_i.empty()) {
+        throw std::logic_error("Parameter -i is not set");
+    }
+
+    if (FLAGS_m.empty()) {
+        throw std::logic_error("Parameter -m is not set");
+    }
+
+    return true;
+}
+
+int main(int argc, char *argv[]) {
+    try {
+        if (!ParseAndCheckCommandLine(argc, argv)) {
+            return 0;
+        }
+            
+        std::vector<std::string> imageNames;
+        parseInputFilesArguments(imageNames);
+
+        // ------------------------------------------Read all images------------------------------------------
+        std::vector<cv::Mat> inputImgs = {};
+        for (const auto & i : imageNames) {
+            const cv::Mat& tmp = cv::imread(i);
+            if (tmp.data == nullptr) {
+                std::cerr << "Could not read image " << i << '\n';
+            } else {
+                inputImgs.push_back(tmp);
+            }
+        }
+
+        // -------------------------------------------Read network--------------------------------------------
+        InferenceEngine::Core ie;
+        if (FLAGS_p_msg) {
+            ie.SetLogCallback(error_listener);
+        }
+        InferenceEngine::CNNNetReader netReader;
+        InferenceEngine::CNNNetwork network;
+        netReader.ReadNetwork(FLAGS_m);
+        std::string binFileName = fileNameNoExt(FLAGS_m) + ".bin";
+        netReader.ReadWeights(binFileName);
+        network = netReader.getNetwork();
+
+        // ----------------------------------------Init inputBlobName-----------------------------------------
+        InferenceEngine::InputsDataMap inputInfo(network.getInputsInfo());
+        if (inputInfo.size() != 1) throw std::logic_error("Sample supports topologies with 1 input only");
+        std::string inputBlobName = inputInfo.begin()->first;
+        unsigned curPos = 0;
+        unsigned framesNum = 0;
+        std::atomic<bool> quitFlag(false);
```

Comment:
These vars don't belong to Init inputBlobName section.

Reply:
Done